examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp

```diff
@@ -95,23 +97,46 @@ EmberAfStatus OTAProviderExample::HandleQueryImage(chip::app::CommandHandler * c
                                                    const chip::app::ConcreteCommandPath & commandPath,
                                                    const QueryImage::DecodableType & commandData)
 {
-    // TODO: add confiuration for returning BUSY status
+    ChipLogDetail(SoftwareUpdate, "Requestor endpoint:%" PRIu16 " node-id:0x" ChipLogFormatX64, commandPath.mEndpointId,
+                  ChipLogValueX64(commandObj->SourceNodeId()));
 
-    OTAQueryStatus queryStatus  = OTAQueryStatus::kNotAvailable;
     uint32_t newSoftwareVersion = commandData.softwareVersion + 1; // This implementation will always indicate that an update is
                                                                    // available (if the user provides a file).
     constexpr char kExampleSoftwareString[] = "Example-Image-V0.1";
-    bool userConsentNeeded                  = false;
     uint8_t updateToken[kUpdateTokenLen]    = { 0 };
     char strBuf[kUpdateTokenStrLen]         = { 0 };
     char uriBuf[kUriMaxLen]                 = { 0 };
+    QueryImageResponse::Type response;
+
+    OTAQueryStatus queryStatus = (strlen(mOTAFilePath) != 0) ? OTAQueryStatus::kUpdateAvailable : OTAQueryStatus::kNotAvailable;
+    bool requestorCanConsent   = commandData.requestorCanConsent.ValueOr(false);
+
+    if (queryStatus == OTAQueryStatus::kUpdateAvailable && mUserConsentDelegate != nullptr)
+    {
+        UserConsentState state = mUserConsentDelegate->GetUserConsentState(commandObj->SourceNodeId(), commandPath.mEndpointId,
+                                                                           commandData.softwareVersion, newSoftwareVersion);
+        switch (state)
+        {
+        case UserConsentState::kGranted:
+            queryStatus = OTAQueryStatus::kUpdateAvailable;
+            break;
 
-    GenerateUpdateToken(updateToken, kUpdateTokenLen);
-    GetUpdateTokenString(ByteSpan(updateToken), strBuf, kUpdateTokenStrLen);
-    ChipLogDetail(SoftwareUpdate, "generated updateToken: %s", strBuf);
+        case chip::ota::UserConsentState::kObtaining:
+            queryStatus = OTAQueryStatus::kBusy;
```

Comment:
Should we set the `delayedActionTime` parameter too then? Otherwise the Requestor won't know when to try again with another QueryImage.

Reply:
Added, PTAL!